### PR TITLE
Adding assertion check to not add backdated journal entries

### PR DESCRIPTION
**Pull request by ranjanikrrao**

### CODE REVIEW

File: pyluca/accountant.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ def enter_journal(
     ):
         if amount == 0:
             return
+        assert not any([entry for entry in self.journal.entries if date < entry.date]), f'Backdated journal entries cannot be added'
```
Comment (sameer-639) [inline pyluca/accountant.py:22]: I guess it should be above amount check

**Review by sameer-639 — APPROVED** [review]

File: pyluca/journal.py [hunk 2]
```diff
@@ -24,3 +24,11 @@ def __init__(
 class Journal:
     def __init__(self, entries: List[JournalEntry] = None):
         self.entries: List[JournalEntry] = [] if entries is None else entries
+        self.max_date: datetime = None if not entries else max([entry.date for entry in entries])
```
Comment (pskd73) [inline pyluca/journal.py:27]: Lets use `... if len(entries) ...` instead?

File: pyluca/tests/test_accountant.py [hunk 3]
```diff
@@ -36,3 +36,24 @@ def test_base(self):
             bal[acct_name] = ledger.get_account_balance(acct_name)
             acct_type_bal[acct['type']] += bal[acct_name]
         self.assertEqual(acct_type_bal['ASSET'], acct_type_bal['INCOME'] - acct_type_bal['EXPENSE'])
+
+    def test_add_journal_entry(self):
+        accountant = Accountant(Journal(), account_config, 'person2')
```
Comment (pskd73) [inline pyluca/tests/test_accountant.py]: We should not initiate Accountant. We are testing Journal, so use only journal

File: pyluca/tests/test_accountant.py [hunk 4]
```diff
@@ -36,3 +36,24 @@ def test_base(self):
...
+        accountant.enter_journal('SAVINGS_BANK', 'SALARY', 30000, datetime(2023, 1, 31), 'Jan salary')
+        self.assertEqual(accountant.journal.max_date, datetime(2023, 1, 31))
+
```
Comment (pskd73) [inline pyluca/tests/test_accountant.py]: Add more assertions for checking the max_date is correctly maintained by adding more entries cumulatively

**Review by pskd73 — CHANGES_REQUESTED** [review]

**Review by pskd73 — APPROVED** [review]

**Pull request merged**